### ISSUE TITLE
Error running rrfs_v1nssl_intel with -DDEBUG=ON using ufs-community:develop

### ISSUE THREAD

**Issue by zach1221** [ufs-community/ufs-weather-model#1944]
## Description

Experience failure when attempting to run rrfs_v1nssl_intel on Hera, with -DDEBUG=ON, against ufs-community:develop branch.

### To Reproduce:

1. log into Hera.
2. Clone ufs-community:dev branch
3. edit rt.conf to add -DDEBUG=ON to compile line of case rrfs_v1nssl_intel
4. Run test

## Additional context

Example from Denise's attempt
/scratch1/NCEPDEV/stmp2/Denise.Worthen/FV3_RT/rt_301756/rrfs_v1nssl_intel
![image](https://github.com/ufs-community/ufs-weather-model/assets/99902696/b9224586-21c5-4228-a854-44eaa21f3b2e)

That line is for bdsnow below

![image](https://github.com/ufs-community/ufs-weather-model/assets/99902696/ff6ef4b8-ac48-43b4-9c80-da871c45ec17)

**Comment by DeniseWorthen** [comment]
Clarification...I think your output above showing the failure in module_mp_nssl_2mom.F90 is the failure that @BrianCurtis-NOAA got w/ the PR #1915 branch on Acorn in non-debug mode. The failure I posted was for debug mode at develop [68050e5](https://github.com/ufs-community/ufs-weather-model/commit/68050e58589a82ab509aaefaafdc2a6b90f34e48)

**Comment by zach1221** [comment]
Running debug with gnu on Hercules. Some failures, but mostly due to timeouts. rrfs_v1beta_gnu seemed to be the only one to fail from another cause. I'll re-test the others that failed with an extended wall clock.                                                                                    working dir  = /work2/noaa/stmp/zshrader/stmp/zshrader/FV3_RT/rt_2728437/  
Logs: /work/noaa/nems/zshrader/hercules/rt-1915/tests/logs/RegressionTests_hercules.log

**Comment by zach1221** [comment]
> Clarification...I think your output above showing the failure in module_mp_nssl_2mom.F90 is the failure that @BrianCurtis-NOAA got w/ the PR #1915 branch on Acorn in non-debug mode. The failure I posted was for debug mode at develop [68050e5](https://github.com/ufs-community/ufs-weather-model/commit/68050e58589a82ab509aaefaafdc2a6b90f34e48)

Yes, that's true. I'll remove it.

**Comment by DeniseWorthen** [comment]
@zach1221 I don't see the ``rrfs_v1nssl`` test in your run directory.

**Comment by zach1221** [comment]
> @zach1221 I don't see the `rrfs_v1nssl` test in your run directory.

@DeniseWorthen it doesn't look like rrfs_v1nssl is part of the gnu tests in rt.conf actually. I can try to compile it with gnu.

**Comment by zach1221** [comment]
It fails with gnu. Screenshot of the err log but I'm not sure it's helpful.
![image](https://github.com/ufs-community/ufs-weather-model/assets/99902696/e880350b-a6ce-410f-82bd-fd7c6ba79270)
/work2/noaa/stmp/zshrader/stmp/zshrader/FV3_RT/rt_2824213/rrfs_v1nssl_gnu

**Comment by BrianCurtis-NOAA** [comment]
> It fails with gnu. Screenshot of the err log but I'm not sure it's helpful. ![image](https://user-images.githubusercontent.com/99902696/275841080-e880350b-a6ce-410f-82bd-fd7c6ba79270.png) /work2/noaa/stmp/zshrader/stmp/zshrader/FV3_RT/rt_2824213/rrfs_v1nssl_gnu

Can't see it well in the SS, but it looks like the glacier file where we see the issue with the intel compiler?

**Comment by zach1221** [comment]
> > It fails with gnu. Screenshot of the err log but I'm not sure it's helpful. ![image](https://user-images.githubusercontent.com/99902696/275841080-e880350b-a6ce-410f-82bd-fd7c6ba79270.png) /work2/noaa/stmp/zshrader/stmp/zshrader/FV3_RT/rt_2824213/rrfs_v1nssl_gnu
> 
> Can't see it well in the SS, but it looks like the glacier file where we see the issue with the intel compiler?

Yes, sorry I cut it off just above the glacier file portion I think.
![image](https://github.com/ufs-community/ufs-weather-model/assets/99902696/a2ac8a62-1551-4c9e-9f1d-942986d1862d)

**Comment by zach1221** [comment]
pointing to the same bdsnow line in module_sf_noahmp_glacier.F90 , that Denise posted previously.

**Comment by zach1221** [comment]
Testing this issue again against ufs-wm:develop.

**Comment by gspetro-NOAA** [comment]
@zach1221 could you offer some insight into the need for running `rrfs_v1nssl_intel` in DEBUG mode? 

I tested `rrfs_v1nssl_intel` in the develop branch (#118cff46) today on Ursa with `-DDEBUG=ON`, and the test still failed despite the updates in PR #1915. The `RegressionTests_ursa.log` file says "FAILED: RUN DID NOT COMPLETE" and the `run_rrfs_v1nssl_intel.log` file says:
```
+ echo 'rt_utils.sh: !!!!!!!!!!JOB TERMINATED!!!!!!!!!! status=FAILED'
rt_utils.sh: !!!!!!!!!!JOB TERMINATED!!!!!!!!!! status=FAILED
```
The `err`  file contains several references to floating invalids. @jkbk2004 has suggested that this indicates a memory issue, either a bug or a resource issue. The test runs fine without DEBUG mode on, so likely a resources issue. 

My run directory is here: `/scratch4/NAGAPE/epic/Gillian.Petro/stmp/RT_RUNDIRS/Gillian.Petro/FV3_RT/rt_1812508` and the test directory is here: /scratch3/NAGAPE/epic/Gillian.Petro/ursa/wm-issue-1944-nssl-debug/tests.

Is this issue still worth pursuing? I'm unclear on why we need to run this test in DEBUG mode.

**Comment by grantfirl** [comment]
@gspetro-NOAA We ran into a similar issue with the SCM using NSSL MP in debug mode. There should be a fix (see https://github.com/NCAR/ccpp-physics/issues/1155) coming into the NCAR branch of ccpp-physics first, then into ufs-dev (and therefore UFS weather model) the next time it is synced back there.

**Comment by gspetro-NOAA** [comment]
@grantfirl Great! Would the update to ufs-dev potentially fix this issue, too, or do you expect that additional steps (beyond the hash update PR) would be needed on our end?

**Comment by grantfirl** [comment]
I think that the code fix for the NSSL physics scheme will be all that's needed for he debug mode to run successfully with the NSSL MP scheme, although I haven't tried it yet. For the SCM, this was all that was required and the SCM shares debug compilation flags with the UFS.

**Comment by gspetro-NOAA** [comment]
@grantfirl Can I close this issue now that PR #2882 has been merged with the NCAR main sync?

**Comment by grantfirl** [comment]
> [@grantfirl](https://github.com/grantfirl) Can I close this issue now that PR [#2882](https://github.com/ufs-community/ufs-weather-model/pull/2882) has been merged with the NCAR main sync?

Yes, this is fixed now.
